src/physics/em/detail/UrbanMscScatter.hh

```diff
@@ -163,48 +177,43 @@ UrbanMscScatter::UrbanMscScatter(const UrbanMscRef&       shared,
 template<class Engine>
 CELER_FUNCTION auto UrbanMscScatter::operator()(Engine& rng) -> MscInteraction
 {
-    MscInteraction result{true_path_, inc_direction_, {0, 0, 0}};
-
-    // Do not sample scattering at the last or at a small step
-    if (true_path_ < helper_.range() && true_path_ > geom_min())
+    if (skip_sampling_)
     {
-        auto end_energy = helper_.end_energy(true_path_);
+        // Do not sample scattering at the last or at a small step
+        return {true_path_, inc_direction_, {0, 0, 0}};
+    }
 
-        bool skip_sampling = (end_energy.value() < 1e-9
-                              || true_path_ <= input_.limit_min
-                              || true_path_ < lambda_ * params_.tau_small);
+    // Ssample azimuthal angle, used for displacement and exiting angle
+    real_type phi
+        = UniformRealDistribution<real_type>(0, 2 * constants::pi)(rng);
 
-        if (!skip_sampling)
+    // Calculate displacement
+    Real3 displacement;
+    if (input_.is_displaced && tau_ >= params_.tau_small)
+    {
+        // Sample displacement and adjust
+        displacement    = this->sample_displacement(rng, phi);
+        real_type rmax2 = (true_path_ - input_.geom_path)
+                          * (true_path_ + input_.geom_path);
+        real_type scaling
+            = this->calc_displacement_scaling(displacement, rmax2);
+        for (auto i : range(3))
         {
-            real_type cth = this->sample_cos_theta(
-                rng, end_energy, input_.true_path, input_.limit_min);
-
-            CELER_ENSURE(std::abs(cth) <= 1);
-
-            // Sample direction
-            UniformRealDistribution<real_type> sample_phi(0, 2 * constants::pi);
-            real_type                          phi = sample_phi(rng);
-            result.direction = rotate(from_spherical(cth, phi), inc_direction_);
-
-            // true_path^2 - geo_path^2
-            real_type rmax2 = (true_path_ - input_.geom_path)
-                              * (true_path_ + input_.geom_path);
-
-            if (input_.is_displaced && tau_ >= params_.tau_small && rmax2 > 0)
-            {
-                // Sample displacement and adjust
-                result.displacement = this->sample_displacement(rng, phi);
-                real_type scaling   = this->calc_displacement_scaling(
-                    result.displacement, rmax2);
-                for (auto i : range(3))
-                {
-                    result.displacement[i] *= scaling;
-                }
-            }
+            displacement[i] *= scaling;
         }
     }
+    else
+    {
+        displacement = {0, 0, 0};
+    }
 
-    return result;
+    // Sample direction
+    real_type mu = this->sample_cos_theta(
```

Comment:
Is mu better than cth? cos_theta? Either way is okay.

Reply:
`$\mu = cos \theta$` is the canonical way of representing polar angle cosines in radiation transport in nuclear engineering... but I guess it's not common in HEP (I don't see it in Geant4). I can revert this usage, but note that there was one function declaration written as `phi` but was called with `cth`.

Reply:
Okay, then let's keep mu - always better to know new/various conventions.